cmd/influxd/run/server_test.go

```diff
@@ -2106,6 +2121,176 @@ func TestServer_Query_AggregatesCommon(t *testing.T) {
 			command: `SELECT COUNT(DISTINCT host) FROM intmany`,
 			exp:     `{"results":[{"series":[{"name":"intmany","columns":["time","count"],"values":[["1970-01-01T00:00:00Z",0]]}]}]}`,
 		},
+	}...)
+
+	for i, query := range test.queries {
+		if i == 0 {
+			if err := test.init(s); err != nil {
+				t.Fatalf("test init failed: %s", err)
+			}
+		}
+		if query.skip {
+			t.Logf("SKIP:: %s", query.name)
+			continue
+		}
+		if err := query.Execute(s); err != nil {
+			t.Error(query.Error(err))
+		} else if !query.success() {
+			t.Error(query.failureMessage())
+		}
+	}
+}
+
+func TestServer_Query_Aggregates_IntMany_GroupBy(t *testing.T) {
+	t.Parallel()
+	s := OpenDefaultServer(NewConfig(), "")
+	defer s.Close()
+
+	test := NewTest("db0", "rp0")
+	test.write = strings.Join([]string{
+		fmt.Sprintf(`intmany,host=server01 value=2.0 %d`, mustParseTime(time.RFC3339Nano, "2000-01-01T00:00:00Z").UnixNano()),
```

Comment:
I've been meaning to write another helper func like `unTime` which would operate like `unTime("2000-01-01T00:00:00Z")` and return the `UnixNano` representation.  the `mustParseTime` is a rather large block of code and makes the the tests hard to read.